rust/processor/src/grpc_stream.rs

```diff
@@ -358,8 +359,13 @@ pub async fn create_fetcher_loop(
     let mut send_ma = MovingAverage::new(3000);
 
     loop {
-        let is_success = match resp_stream.next().await {
-            Some(Ok(mut r)) => {
+        let is_success = match tokio::time::timeout(
+            indexer_grpc_response_item_timeout_secs,
+            resp_stream.next(),
+        )
+        .await
+        {
+            Ok(Some(Ok(mut r))) => {
```

Comment:
could you please add comments as to what state each of these rows represents? Could also break out into a nested match, up to you